lib/ImageHelper.php

```diff
@@ -407,9 +429,61 @@ public static function sideload_image( $file ) {
 		$file = wp_upload_bits($locinfo['basename'], null, file_get_contents($file_array['tmp_name']));
 		// delete tmp file
 		@unlink($file_array['tmp_name']);
+
+		remove_filter( 'upload_dir', [ __CLASS__, 'set_sideload_image_upload_dir' ] );
+
 		return $file['url'];
 	}
 
+	/**
+	 *
+	 * Used by ImageHelper::sideload_image().
+	 *
+	 * @internal
+	 * @since 2.0.0
+	 * @see   \Timber\ImageHelper::sideload_image()
+	 *
+	 * @param array $upload Array of information about the upload directory.
+	 *
+	 * @return array 		Array of information about the upload directory, modified by this 
```

Comment:
Originally, this was `mixed`. This was my most substantive change — want to make sure that's correct!

Reply:
@jarednova I’m not sure I follow you here. Which parameter was `mixed`?

Reply:
Sorry, should have been more clear. The `@return` param was originally `mixed`.

Reply:
You mean for the `upload_dir` filter in WordPress, or in Timber? Because I didn’t actually change a parameter type, or did I?

Reply:
In Timber, the (new?) method for `ImageHelper:: set_sideload_image_upload_dir` had a return type of `mixed`. I guess conceivably it _could_ be mixed if there's an error or given a non-array value from a different function that's also hooked into `upload_dir`

Reply:
Hm... when did it have that? I guess I still don’t where exactly it had or could have had a `mixed` return type. It’s a function that is going be first introduced with this pull request, right? I’m sorry if I still don’t understand 😅.

When I look at the `upload_dir` filter, it specifically requests an `array` type that contains certain values and can’t be `mixed` either. If there would be an error, it should be inside the array with the `error` key.

```php
/**
 * Filters the uploads directory data.
 *
 * @since 2.0.0
 *
 * @param array $uploads {
 *     Array of information about the upload directory.
 *
 *     @type string       $path    Base directory and subdirectory or full path to upload directory.
 *     @type string       $url     Base URL and subdirectory or absolute URL to upload directory.
 *     @type string       $subdir  Subdirectory if uploads use year/month folders option is on.
 *     @type string       $basedir Path without subdir.
 *     @type string       $baseurl URL path without subdir.
 *     @type string|false $error   False or error message.
 * }
 */
$uploads = apply_filters( 'upload_dir', $cache[ $key ] );
```

Reply:
Yeah, I think we're good here. For posterity, you can see from this [prior commit](https://github.com/timber/timber/blob/302acbfdc7faef8998d92620a9f6ebb583943cad/lib/ImageHelper.php#L448) where it was `mixed` earlier in this PR. But regardless, it's correct as it stands right now so good to merge!

Reply:
Ooooh, now I see! Thanks for searching for that older commit. I think that `mixed` might have been dropped in through code completion in PhpStorm and I corrected it to `array` in a later commit.